### PR TITLE
Tests for core settings code

### PR DESCRIPTION
**Pull request by kirkbushell**
This is the start of something beautiful - TESTS!

@franzliedke had already added some tests using phpspec, but due to some of the complexity and use-cases (such as the repository object chaining), said tool wasn't going to work for all unit/behavioural tests. As a result, and after a discussion with Franz, have added phpunit, mockery, and converted the existing specs to unit tests.

This should act as a pattern/approach for the rest of the project, and I'll begin by implementing this for the settings handlers and other associated code before moving onto other modules.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,19 +36,21 @@
...
+        "mockery/mockery": "^0.9.4",
         "squizlabs/php_codesniffer": "2.*",
-        "phpspec/phpspec": "^2.2"
+        "phpunit/phpunit": "^4.8"
     },
     "autoload": {
         "psr-4": {
-            "Flarum\\": "src/"
+            "Flarum\\": "src/",
+            "tests\\": "tests/"
```
Comment (franzliedke) [inline composer.json]: Please move this line to an `autoload-dev` block just like this one (it has the same structure)...

Comment (kirkbushell) [inline composer.json]: Ah! That's what you meant :)

Fixed.

**Comment by franzliedke** [comment]
Thank you so much!

**Comment by tobscure** [comment]
> This is the start of something beautiful - TESTS!

:sparkling_heart:

**Pull request merged**